spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/main/java/org/springframework/boot/configurationprocessor/metadata/JsonConverter.java

```diff
@@ -160,26 +160,18 @@ private Object extractItemValue(Object value) {
 
 	private static class ItemMetadataComparator implements Comparator<ItemMetadata> {
 
+		private final Comparator<ItemMetadata> itemComparator = Comparator.comparing(this::isDeprecated)
+				.thenComparing(ItemMetadata::getName).thenComparing(ItemMetadata::getSourceType);
+
+		private final Comparator<ItemMetadata> groupComparator = Comparator.comparing(ItemMetadata::getName)
+				.thenComparing(ItemMetadata::getSourceType);
+
 		@Override
 		public int compare(ItemMetadata o1, ItemMetadata o2) {
 			if (o1.isOfItemType(ItemType.GROUP)) {
-				return compareGroup(o1, o2);
-			}
-			return compareProperty(o1, o2);
-		}
-
-		private int compareGroup(ItemMetadata o1, ItemMetadata o2) {
-			return o1.getName().compareTo(o2.getName());
-		}
-
-		private int compareProperty(ItemMetadata o1, ItemMetadata o2) {
-			if (isDeprecated(o1) && !isDeprecated(o2)) {
-				return 1;
-			}
-			if (isDeprecated(o2) && !isDeprecated(o1)) {
-				return -1;
+				return this.groupComparator.compare(o1, o2);
```

Comment:
> This is not possible. Sort is applied after filtering of items to have just one type:

Sorry to keep bothering you. In the case of `GROUP` type, `this::is Deprecated` will always be false. Isn't it okay to use `itemComparator` regardless of type?

Reply:
I don't use it for items, only for groups - take a look at `compare(ItemMetadata o1, ItemMetadata 02)`, diff can be misleading